### PR TITLE
Fix code editor's ColorPicker issue

### LINKED ISSUE #63880: Color picker in code editor is broken when editing color constant
### Godot version

4.0 4ac993ffe2

### System information

Mint 10

### Issue description

Color picker in editor is broken while editing color constants. It replaces whole line with color params.

### Steps to reproduce

1. Have piece of code with color, in my case:
```
func _draw() -> void:
	draw_circle(_vertex, _radius, Color.RED)
```
2. Right click on color, choose color picker from context menu
3. Change color
4. Instead of replacing color, whole line is replaced with color params:
```
func _draw() -> void:
(0.7421875, 0.7421875, 0.7421875)
```
As a bonus you can't revert this operation

### Minimal reproduction project

_No response_

### PR DESCRIPTION
**Pull request by heppocogne**
This commit fixes #63880 
- Fix an issue that caused editor replacing whole the line with color parameter on editing a color constant.
- If the color constant does not exist, "Pick Color" is not shown in the context menu.

image
image

- "Pick Color" is no longer shown in the context menu if  `Color`'s parameter contains brackets such as `Color((0),0,0)` because this expression also caused the same problem.
image

### CODE REVIEW

File: editor/plugins/script_text_editor.cpp [hunk 1]
```diff
@@ -1715,30 +1715,67 @@ void ScriptTextEditor::_text_edit_gui_input(const Ref<InputEvent> &ev) {
 			color_position.x = row;
 			color_position.y = col;
 
-			int begin = 0;
-			int end = 0;
-			bool valid = false;
+			int begin = -1;
+			int end = -1;
+			int expression_pattern = 0;
```
Comment (JohanAR) [inline editor/plugins/script_text_editor.cpp:1720]: Wouldn't this be better as an enum?

Comment (heppocogne) [reply editor/plugins/script_text_editor.cpp]: Thanks for the review.
I have modified the code according to your suggestion.

Comment (JohanAR) [reply editor/plugins/script_text_editor.cpp]: Looks good IMO, easier to read and understand now. I'm not a member of the Godot team so I can't officially review or accept PRs

**Comment by iwek7** [comment]
I've tested this and it works great!

**Comment by KoBeWi** [comment]
Needs rebase.

**Review by KoBeWi — APPROVED** [review]
Ok I rebased it myself. Original patch for the record:
[63907.patch.txt](https://github.com/godotengine/godot/files/10711861/63907.patch.txt)

The PR fixes the issue and code looks ok (local enum declaration is odd though; I didn't even know it's possible).

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**